gomint-server/src/main/java/io/gomint/server/network/handler/PacketLoginHandler.java

```diff
@@ -147,171 +139,170 @@ public void handle(PacketLogin packet, long currentTimeMillis, PlayerConnection
                 validSkin = false;
             }
 
-            // Sync up for disconnecting etc.
-            boolean finalValidSkin = validSkin;
-            connection.server().syncTaskManager().addTask(new SyncScheduledTask(connection.server().syncTaskManager(), () -> {
-                // Invalid skin
-                if (!finalValidSkin) {
-                    connection.disconnect("Skin is invalid or corrupted");
-                    return;
-                }
+            // Invalid skin
+            if (!validSkin) {
+                connection.disconnect("Skin is invalid or corrupted");
```

Comment:
disconnect doesn't work since there is a packet sending safeguard but no world is set. This would NPE in packet sending

Reply:
I don't see any world safeguard in packet sending

Reply:
https://github.com/gomint/gomint/pull/727/files#diff-59e6fa0629aae961d769bd023789e9fe6686ceb54bda8fec9f4b3f216c33cddbR257 This would fire warnings when dropping a user connection in the packet login handler

Reply:
I guess we can just also check world for null then there

Reply:
Are packet handlers invoked from the world thread where the player is in? Otherwise stuff like settings request will also generate that warning. If they do a `!= null` check seems good

Reply:
resolved?